### PR TITLE
OBPIH-4047 Fix issues with invoice item candidate not appearing after…

### PR DESCRIPTION
**Pull request by dbienkowska**
… first not full invoice

### CODE REVIEW

File: grails-app/migrations/views/invoice-candidate-view.sql [hunk 1]
```diff
@@ -11,6 +11,7 @@ CREATE OR REPLACE VIEW invoice_item_candidate AS
         order_adjustment.description as description,
         1 as quantity,
         1 as quantity_to_invoice,
+        0 as quantity_from_invoice,
```
Comment (awalkowiak) [inline grails-app/migrations/views/invoice-candidate-view.sql:14]: Is this used somewhere?

File: grails-app/migrations/views/invoice-candidate-view.sql [hunk 2]
```diff
@@ -38,8 +39,11 @@ union
         product.gl_account_id as gl_account_id,
         product.product_code as product_code,
         product.name as description,
-        shipment_item.quantity/order_item.quantity_per_uom as quantity,
-        shipment_item.quantity/order_item.quantity_per_uom as quantity_to_invoice,
+        case when invoice_item.id is not null then (shipment_item.quantity/order_item.quantity_per_uom) - invoice_item.quantity
+        else shipment_item.quantity/order_item.quantity_per_uom end as quantity,
+	    case when invoice_item.id is not null then (shipment_item.quantity/order_item.quantity_per_uom) - invoice_item.quantity
+        else shipment_item.quantity/order_item.quantity_per_uom end as quantity_to_invoice,
+        invoice_item.quantity as quantity_from_invoice,
```
Comment (awalkowiak) [inline grails-app/migrations/views/invoice-candidate-view.sql:46]: same question as above

File: grails-app/migrations/views/invoice-candidate-view.sql [hunk 3]
```diff
@@ -38,8 +39,10 @@ union
         product.gl_account_id as gl_account_id,
         product.product_code as product_code,
         product.name as description,
-        shipment_item.quantity/order_item.quantity_per_uom as quantity,
-        shipment_item.quantity/order_item.quantity_per_uom as quantity_to_invoice,
+        case when invoice_item.id is not null then (shipment_item.quantity/order_item.quantity_per_uom) - invoice_item.quantity
```
Comment (awalkowiak) [inline grails-app/migrations/views/invoice-candidate-view.sql:42]: @dbienkowska is it possible, that the shipment item will have more than two invoice items? then it should be sum of all `invoice_item.quantity`

Comment (awalkowiak) [reply grails-app/migrations/views/invoice-candidate-view.sql:41]: @dbienkowska Additionally, do we need both `quantity` and `quantity_to_invoice`? Seems like these values are the same.

**Review by jmiranda — APPROVED** [review]
LCTM - looks complicated to me

**Review by pmuchowski — APPROVED** [review]

**Pull request merged**